clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -2834,9 +2834,15 @@ static void ParseLangArgs(LangOptions &Opts, ArgList &Args, InputKind IK,
     }
   }
 
+  Opts.SYCLIsDevice   = Args.hasArg(options::OPT_fsycl_is_device);
+  Opts.SYCLXOCCDevice = Args.hasArg(options::OPT_fsycl_xocc_device);
+  Opts.SYCLAllowFuncPtr = Args.hasFlag(options::OPT_fsycl_allow_func_ptr,
+                                  options::OPT_fno_sycl_allow_func_ptr, false);
```

Comment:
Not even close unfortunately, plus that's the original indentation from the Intel implementation in this case, so I think it should be fine to leave it as is for now. Makes it more difficult to compare for differences when merging otherwise I think